### PR TITLE
Add a proper entry point for the docker image

### PR DESCRIPTION
**Pull request by MJafarMashhadi**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,6 +33,44 @@ WORKDIR /app
...
+    && apt-get purge -y --auto-remove $buildDeps \
+    && find /usr/local -depth \
+    \( \
+        \( -type d -a -name test -o -name tests \) \
+        -o \
+        \( -type f -a -name '*.pyc' -o -name '*.pyo' \) \
+    \) -exec rm -rf '{}' +
+
+FROM alchemy-with-code as staging
 RUN pip install -r requirements/production.txt
```
Comment (jzhang-gp) [inline Dockerfile:74]: I'm a bit confused. Are we not installing the production requirements into the `production` stage?

File: Dockerfile [hunk 2]
```diff
@@ -20,6 +20,9 @@ ENV PATH $PATH:/root/tools/google-cloud-sdk/bin
 # This is necessary for the alembic migration to work
 ENV PYTHONPATH /app
 
+ENTRYPOINT ["ci/run_server.sh"]
+CMD ["admin_server"]
```
Comment (jzhang-gp) [inline Dockerfile:24]: Is the default parameter necessary?

**Review by jzhang-gp — COMMENTED** [review]
Have you tested the server is up and running (create a task, do a few annotations and train a model) since this involves changes on how we run production server?

File: Dockerfile (hunk 2 above)
Comment (MJafarMashhadi) [reply Dockerfile:24]: The CMD line is not very necessary; the ENTRYPOINT command should be there though

File: Dockerfile (hunk 1 above)
Comment (MJafarMashhadi) [reply Dockerfile]: Yes, we are installing production requirements in production stage.
uwsgi needs a C compiler to install, these commands install `gcc`, `make`, etc before running `pip install -r requirements/production.txt` and after that, it removes them all.

**Comment by MJafarMashhadi** [comment]
> Have you tested the server is up and running (create a task, do a few annotations and train a model) since this involves changes on how we run production server?

Yes, it works :)

**Comment by MJafarMashhadi** [comment]
> Have you tested the server is up and running (create a task, do a few annotations and train a model) since this involves changes on how we run production server?

It works, yes

File: Dockerfile (hunk 1 above)
Comment (jzhang-gp) [reply Dockerfile]: Ah I see. It's mixed with the rest of the command.

File: Dockerfile (hunk 2 above)
Comment (jzhang-gp) [reply Dockerfile:24]: Can we remove it? I feel this is a little bit confusing.

**Review by jzhang-gp — APPROVED** [review]
Just a minor comment on the `CMD` in the dockerfile. The rest looks fine.

File: Dockerfile (hunk 2 above)
Comment (MJafarMashhadi) [reply Dockerfile]: Done :)

**Pull request merged**